base/ltcounts.dtx

```diff
@@ -371,7 +374,9 @@
 %<latexrelease>                             {provide theHfoo commands}%
 %<*2ekernel|latexrelease>
 \def\@addtoreset#1#2{\expandafter\@cons\csname cl@#2\endcsname {{#1}}%
-  \expandafter\gdef\csname theH#1\endcsname{\csname theH#2\endcsname.\the\value{#1}}%
+  \expandafter\xdef\csname theH#1\endcsname{%
+    \expandafter\noexpand\csname theH#2\endcsname.%
+    \noexpand\the\noexpand\value{#1}}%
```

Comment:
better perhaps?

\expandafter\gdef\csname theH#1\expandafter\endcsname\expandafter{\csname theH#2\endcsname.\the\value{#1}}%

Reply:
or maybe not in case `#1` also needs expanding

Reply:
ah, I see the problem, I used \value to avoid that weird redefinitions of `\@arabic`  break the pdfstrings, but then one no longer has the expansion from the \csname. The xdef is imho fine.

Reply:
> or maybe not in case `#1` also needs expanding

@FrankMittelbach The main point here is that `#1` needs expanding. The one-step expansion on `\csname theH#2\endcsname` is just an incidental result.

An alternative solution is to fully expand both counter names (in `\newcounter{<ctr name1>}[<ctr name2>]`) as early as possible, but that needs redefine more commands. Or, are the raw, unexpanded counter names used in any error messages?